### PR TITLE
Add planned enhancement for weight and more specific error messages for optional fields

### LINKED ISSUE #276: [PE-D][Tester C] `Edit` command: Standardise behaviour for empty weight and empty height given
Problem: `Edit` command gives different behaviour when empty weight and empty height is given (in separate `edit` command)

Steps to reproduce: (from empty app)
1. `add n/John p/123`
1. `edit 1 w/`
1. 'edit 1 h/'

`edit 1 w/` when client has no weight information outputs error message: "There are no more weight values to be removed. This client has no more weight values associated with them."
`edit 2 w/` when client has no height information succeeds.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: shaunlxw/ped#14

### LINKED ISSUE #300: [PE-D][Tester A] Tracking of clients' weight feature is incomplete
To edit the weight of a client from a day in the past, the weight of the clients from the day onwards have to be all removed, causing inconvenience to the users especially if they do not want the correct weights to be deleted. Furthermore new weights added cannot be assigned to previous dates, causing inconvenience to users who may forget to add in the weight of their client.
**UG's explanation of editing weight:**

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Lalelulilulela/ped#9

### PR DESCRIPTION
**Pull request by ruijietay**
Resolves #300, resolves #276.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -632,3 +632,10 @@ testers are expected to do more *exploratory* testing.
     1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
 
 1. _{ more test cases …​ }_
+
+
+## **Appendix: Planned Enhancements**
+FitBook's team size is 5.
+
+1. **Allow for more flexible weight management:** The current implementation of tracking client's weights only allow users to add or modify the latest weight value at the current date and time. They can also delete the latest weight value from the client. We plan to make this feature more flexible by allowing users to add/modify a client's weight at specified date and time of their choice.
+1. **Provide more specific error messages:** When removing non-optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages to the other non-optional fields of a client , where the error message will be similar to the one seen in the `weight` field.
```
Comment (ruijietay) [inline docs/DeveloperGuide.md:641]:
```suggestion
1. **Provide more specific error messages:** When removing non-optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages to the other optional fields of a client , where the error message will be similar to the one seen in the `weight` field.
```
This is a typo!

**Review by ruijietay — COMMENTED** [review]
Typo to fix

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -632,3 +632,10 @@ testers are expected to do more *exploratory* testing.
     1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
 
 1. _{ more test cases …​ }_
+
+
+## **Appendix: Planned Enhancements**
+FitBook's team size is 5.
+
+1. **Allow for more flexible weight management:** The current implementation of tracking client's weights only allow users to add or modify the latest weight value at the current date and time. They can also delete the latest weight value from the client. We plan to make this feature more flexible by allowing users to add/modify a client's weight at specified date and time of their choice.
+1. **Provide more specific error messages:** When removing non-optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages to the other optional fields of a client , where the error message will be similar to the one seen in the `weight` field.
```
Comment (ruijietay) [inline docs/DeveloperGuide.md:641]:
```suggestion
1. **Provide more specific error messages:** When removing optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages when a client edits a field that has not changed / has nothing to remove, where the error message will be similar to the one seen in the `weight` field.
```

**Review by dillontkh — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -632,3 +632,10 @@ testers are expected to do more *exploratory* testing.
     1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
 
 1. _{ more test cases …​ }_
+
+
+## Appendix: Planned Enhancements
+FitBook's team size is 5.
+
+1. **Allow for more flexible weight management:** The current implementation of tracking clients' weights only allows users to add, modify or delete the latest weight value at the current date and time. We plan to make this feature more flexible by allowing users to add/modify a client's weight at a specified date and time of their choice.
+1. **Provide more specific error messages:** When removing optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages when a client edits a field that has not changed / has nothing to remove, where the error message will be similar to the one seen in the `weight` field.
```
Comment (ruijietay) [inline docs/DeveloperGuide.md:641]:
```suggestion
1. **Provide more specific error messages:** When removing optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages when a client edits a field that has not changed / has nothing to remove, where the error message will be similar to the one seen in the `weight` field.

```

**Review by RyanNgWH — APPROVED** [review]
LGTM

**Pull request merged**